### PR TITLE
Plans overhaul: Starter Plan for the win

### PR DESCRIPTION
**Pull request by chriskmnds**
### Changes proposed in this Pull Request

Addresses 712-gh-Automattic/martech

- Adds the Starter plan to the plans grid in place of the Free plan at `/plans` and `/start/plans`.
- Button & header text based on pdgrnI-WP-p2#comment-1737

Requires backend patch: D80430-code

#### `/start/plans` (with free domain)

Screenshot 2022-05-11 at 11 54 12 AM

#### `/plans` (with site on Starter plan)

Screenshot 2022-05-10 at 4 35 55 PM

### Testing instructions

**with feature flag set `?flags=plans/starter-plan`**
- Apply D80430-code
- Go to `/start/domains` and do not select a custom/paid domain
    - In `/start/plans` next:
        - Confirm Starter is shown in place of the Free plan, matching the design mock (pdgrnI-WP-p2) and media above.
        - Confirm the link is shown in the header for purchasing a Free plan instead.
        - Click on "Start with Starter" and confirm checkout works (Starter plan in cart next). Should be able to purchase the Starter plan as normal here too.
- Go to `/start/domains` and select a custom/paid domain
    - In `/start/plans` next:
        - Confirm No link to the plan is shown in header.
- Go to `/plans`
        - Confirm Starter is shown in place of the Free plan, matching the design mock (pdgrnI-WP-p2) and media above

**without feature flag set (disable with `-`) `?flags=-plans/starter-plan`**
- Confirm both `/start/plans` and `/plans` show no Starter and the UI remains the same as in `trunk` branch

### TODO
- [x] ~Confirm and update price displays (there are decimals in the design mocks e.g. `$5.00` pdgrnI-WP-p2)~ (moved to https://github.com/Automattic/wp-calypso/issues/63501)
- [x] ~Confirm and update including yearly price in plan description (`per month, billed $60 yearly` in design mocks pdgrnI-WP-p2)~ (moved to https://github.com/Automattic/wp-calypso/issues/63501)
- [x] ~Update My Plan page for Starter with icon and features. Use Personal plan's~ (moved to https://github.com/Automattic/wp-calypso/issues/63499)
- [x] Some state/data missing:
    - to update the Starter plan price in the grid (currently shows 0) 
    - to match a site on the Starter plan with the various UI elements in `/plans`
        - shows "free" in place of the current plan
        - doesn't update the plans grid to show "This is your plan" button

Fixes Automattic/martech#712

### CODE REVIEW

File: client/my-sites/plans-comparison/plans-comparison-features.tsx [hunk 1]
```diff
@@ -125,6 +125,144 @@ export const planComparisonFeatures: PlanComparisonFeature[] = [
...
+	{
+		get title() {
+			return translate( 'Storage' );
+		},
+		get description() {
+			return translate(
+				'The Starter plan allows a maximum storage of 6GB, which equals to approximately 1200 high quality images. With WordPress Pro you may go all the way up to 50GB, enough space for 10,000 high quality images of the same size.'
+			);
+		},
+		features: [ FEATURE_6GB_STORAGE, FEATURE_50GB_STORAGE ],
```
Comment (chriskmnds) [inline client/my-sites/plans-comparison/plans-comparison-features.tsx]: Note to self: `FEATURE_1GB_STORAGE` ought to be here somewhere

File: client/my-sites/plans/main.jsx [hunk 2]
```diff
@@ -120,21 +120,10 @@ class Plans extends Component {
...
 		const checkoutPath = `/checkout/${ selectedSite.slug }/${ item.product_slug }/`;
 
-		page(
-			discount
-				? addQueryArgs( checkoutPath, {
-						coupon: discount,
-				  } )
-				: checkoutPath
-		);
+		page( checkoutPath );
```
Comment (chriskmnds) [inline client/my-sites/plans/main.jsx]: undo❓ 🧹

File: client/my-sites/plans-comparison/plans-comparison-features.tsx [hunk 3]
```diff
@@ -125,6 +127,153 @@ export const planComparisonFeatures: PlanComparisonFeature[] = [
...
+			if ( feature === FEATURE_50GB_STORAGE ) {
+				storageSize = '50';
+			}
+
+			if ( isMobile ) {
+				if (
+					isLegacySiteWithHigherLimits &&
+					feature &&
+					[ FEATURE_1GB_STORAGE, FEATURE_6GB_STORAGE ].includes( feature ) &&
+					Number( legacyStorageSize ) > Number( storageSize )
```
Comment (chriskmnds) [inline client/my-sites/plans-comparison/plans-comparison-features.tsx]: questionable logic here

File: client/signup/steps/plans/index.jsx [hunk 4]
```diff
@@ -316,6 +319,18 @@ export class PlansStep extends Component {
 		const { hideFreePlan, subHeaderText, translate, eligibleForProPlan, locale } = this.props;
 
 		if ( eligibleForProPlan ) {
+			if ( isStarterPlanEnabled() ) {
+				return translate(
+					'Try risk-free with a 14-day money back guarantee.{{free}}Do you want a free plan? {{link}}Click here{{/link}}.{{/free}}',
+					{
+						components: {
+							free: hideFreePlan ? null : <p />,
```
Comment (chriskmnds) [inline client/signup/steps/plans/index.jsx]: Fix. cannot nest `<p>` inside `<p>`

File: client/my-sites/plans-comparison/use-plan-prices.tsx [hunk 5]
```diff
@@ -11,24 +11,29 @@ function toMonthlyPrice( yearlyPrice: number ) {
...
+			const productId = plan.getProductId();
 
-		return [ getPlanRawPrice( state, productId ), getDiscountedRawPrice( state, productId ) ].map(
-			toMonthlyPrice
-		);
-	} );
+			// @todo clk
+			// if ( plan.type === TYPE_FREE || plan.type === TYPE_FLEXIBLE ) {
+			// 	return [ 0 ];
+			// }
```
Comment (chriskmnds) [inline client/my-sites/plans-comparison/use-plan-prices.tsx]: Needs to revisit why this was needed

File: client/my-sites/plans/navigation.jsx [hunk 6]
```diff
@@ -86,6 +86,7 @@ export default connect( ( state ) => {
 	let isFreeOrFlexible = false;
 
 	if ( eligibleForProPlan && currentPlan ) {
+		// @todo clk
```
Comment (chriskmnds) [inline client/my-sites/plans/navigation.jsx]: Not sure what this is either or whether we need similar conditioning for the Starter plan

File: client/signup/steps/plans/index.jsx [hunk 7]
```diff
@@ -316,6 +319,19 @@ export class PlansStep extends Component {
 		const { hideFreePlan, subHeaderText, translate, eligibleForProPlan, locale } = this.props;
 
 		if ( eligibleForProPlan ) {
+			if ( isStarterPlanEnabled() ) {
+				return translate(
+					'Try risk-free with a 14-day money back guarantee.{{free}}{{break/}}Do you want a free plan? {{link}}Click here{{/link}}.{{/free}}',
```
Comment (chriskmnds) [inline client/signup/steps/plans/index.jsx]: If there's a better and cleaner way to not use `<br />` either

File: packages/calypso-products/src/plans-list.tsx [hunk 8]
```diff
@@ -1574,6 +1576,27 @@ PLANS_LIST[ PLAN_P2_FREE ] = {
...
+	getDescription: () =>
+		i18n.translate( 'Start your WordPress.com website. Limited functionality and storage.' ),
+	getBillingTimeFrame: () => i18n.translate( 'per month, billed yearly' ),
+	getPlanCompareFeatures: () => [
+		FEATURE_CUSTOM_DOMAIN,
+		FEATURE_UNLIMITED_ADMINS,
+		FEATURE_6GB_STORAGE,
+	],
+	getIncludedFeatures: () => [], // @todo clk ?
+	getInferiorFeatures: () => [],
```
Comment (chriskmnds) [inline packages/calypso-products/src/plans-list.tsx]: Not sure what there are 👽

File: packages/calypso-products/src/product-values.ts [hunk 9]
```diff
@@ -15,6 +15,7 @@ export { isBlogger } from './is-blogger';
 export { isBundled } from './is-bundled';
 export { isBusiness } from './is-business';
 export { isPro } from './is-pro';
+export { isStarter } from './is-starter';
```
Comment (chriskmnds) [inline packages/calypso-products/src/product-values.ts:18]: Not used anywhere. could remove, or keep as there's similar for every other plan

File: packages/calypso-products/src/constants/types.ts [hunk 10]
```diff
@@ -12,6 +12,7 @@ export const TYPE_ALL = 'TYPE_ALL';
 export const TYPE_P2_PLUS = 'TYPE_P2_PLUS';
 export const TYPE_FLEXIBLE = 'TYPE_FLEXIBLE';
 export const TYPE_PRO = 'TYPE_PRO';
+export const TYPE_STARTER = 'TYPE_STARTER'; // @todo find use
```
Comment (chriskmnds) [inline packages/calypso-products/src/constants/types.ts]: Weird we have these types and also the other product constants in [wpcom.ts](https://github.com/Automattic/wp-calypso/blob/update/plans-overhaul-starter/packages/calypso-products/src/constants/wpcom.ts). I think i can tell what they refer to, but still, having both seems redundant. Also, unless mistaken, we condition on both at places for same purpose e.g. in [plans-comparison-action.tsx](https://github.com/Automattic/wp-calypso/blob/update/plans-overhaul-starter/client/my-sites/plans-comparison/plans-comparison-action.tsx)?

File: packages/calypso-products/src/product-values.ts (hunk 9 above)
Comment (claudiucelfilip) [reply packages/calypso-products/src/product-values.ts:18]: I think this will be needed for the upcoming Starter Thank You page #63446.

File: client/my-sites/plans/navigation.jsx (hunk 6 above)
Comment (claudiucelfilip) [reply client/my-sites/plans/navigation.jsx]: This part has gone through two changes for the Pro plan overhaul:
- only show the Plans tab if the user is on a Free plan and don't show the Plans tab if the user is on a Pro plan #61656
- show both tabs if the user is on any legacy plan #61981

I'd imagine we would show both tabs for the Starter plan (as is currently).

**Comment by claudiucelfilip** [comment]
I went through the steps and code and didn't find any issues.

For the My Plan page, however, we might want to add a Plan icon and some features (maybe similar to the  Personal plan?):
Screenshot on 2022-05-11 at 11-51-00

File: client/signup/steps/plans/index.jsx [hunk 11]
```diff
@@ -316,6 +319,19 @@ export class PlansStep extends Component {
...
+				return hideFreePlan
+					? translate( 'Try risk-free with a 14-day money-back guarantee.' )
+					: translate(
+							'Try risk-free with a 14-day money-back guarantee or {{link}}start with a free site{{/link}}.',
+							{
+								components: {
+									link: <Button onClick={ this.handleFreePlanButtonClick } borderless={ true } />,
+								},
+							}
+					  );
```
Comment (chriskmnds) [inline client/signup/steps/plans/index.jsx:255]: Need to come back to this. These are two separate strings and not sure if ideal/preferred. I seem to have had trouble getting the final period rendering with something like:

```
				return translate(
					'Try risk-free with a 14-day money-back guarantee{{free}} or {{link}}start with a free site{{/link}}{{/free}}.',
					{
						components: {
							free: hideFreePlan ? null : <span />,
							link: <Button onClick={ this.handleFreePlanButtonClick } borderless={ true } />,
						},
					}
				);
```

guessing `null` not acting friendly 😀

File: packages/calypso-products/src/constants/types.ts (hunk 10 above)
Comment (chriskmnds) [reply packages/calypso-products/src/constants/types.ts]: Moved to issue https://github.com/Automattic/wp-calypso/issues/63497

File: client/my-sites/plans/navigation.jsx (hunk 6 above)
Comment (chriskmnds) [reply client/my-sites/plans/navigation.jsx]: thanks for explaining :-)

**Comment by chriskmnds** [comment]
> I went through the steps and code and didn't find any issues.
> 
> For the My Plan page, however, we might want to add a Plan icon and some features (maybe similar to the Personal plan?): Screenshot on 2022-05-11 at 11-51-00

Thanks for testing. Moved this to an issue. https://github.com/Automattic/wp-calypso/issues/63499

**Comment by chriskmnds** [comment]
We need to confirm if the Plan prices ought to include decimals and yearly sums in the description (as per the design mocks):

- Confirm and update price displays (there are decimals in the design mocks e.g. $5.00 pdgrnI-WP-p2)
- Confirm and update including yearly price in plan description (per month, billed $60 yearly in design mocks pdgrnI-WP-p2)

^ Moved this too to an issue. https://github.com/Automattic/wp-calypso/issues/63501

File: client/lib/plugins/utils.js [hunk 12]
```diff
@@ -338,6 +338,7 @@ export function getPluginAuthorProfileKeyword( plugin ) {
  * @returns the correct business plan slug depending on current plan and pluginBillingPeriod
  */
 export function businessPlanToAdd( currentPlan, pluginBillingPeriod, eligibleForProPlan ) {
+	// @todo clk any rule for Starter plan?
```
Comment (chriskmnds) [inline client/lib/plugins/utils.js]: @claudiucelfilip familiar with these parts?

Comment (claudiucelfilip) [reply client/lib/plugins/utils.js]: Nope, I'm not.

File: client/my-sites/plans-comparison/plans-comparison-action.tsx [hunk 13]
```diff
@@ -88,7 +92,7 @@ export const PlansComparisonAction: React.FunctionComponent< Props > = ( {
 		manageHref = undefined;
 	}
 
-	if ( ! isInSignup && [ TYPE_FLEXIBLE, TYPE_FREE ].includes( plan.type ) ) {
+	if ( ! isInSignup && [ TYPE_FLEXIBLE, TYPE_FREE, TYPE_STARTER ].includes( plan.type ) ) {
```
Comment (claudiucelfilip) [inline client/my-sites/plans-comparison/plans-comparison-action.tsx]: This will not show the `Choose Starter` button on the Plans page if you're on a Free plan site.
Screenshot on 2022-05-11 at 16-45-30

Is this intentional?

Comment (chriskmnds) [reply client/my-sites/plans-comparison/plans-comparison-action.tsx]: Good catch. I don't see why it would need to be like that. 

Updated. Mind checking again?

Comment (chriskmnds) [reply client/my-sites/plans-comparison/plans-comparison-action.tsx]: but let's cc @southp -> free plans can upgrade to Starter right?

**Review by claudiucelfilip — APPROVED** [review]
Thanks for the fixes!

Retested. LGTM 🚀

File: client/lib/plugins/utils.js (hunk 12 above)
Comment (chriskmnds) [reply client/lib/plugins/utils.js]: Something to do with plugin install buttons, adding a required business plan to checkout. I'm guessing `eligibleForProPlan` picks up and adds the Pro plan slug, so probably no need for further action here.

**Comment by chriskmnds** [comment]
Noticed a bit of a bug with the storage limits display, but hopefully it's fixed now (if I even understand correctly). It will/should not cross out anything if the legacy limit is not higher than the new plan limit (as with Pro). I probably messed up earlier, thinking we'd always want to highlight the difference.

**Before**

Screenshot 2022-05-12 at 4 54 25 PM

**After**

Screenshot 2022-05-12 at 4 53 07 PM

File: client/blocks/plan-storage/index.jsx [hunk 14]
```diff
@@ -60,9 +61,14 @@ export function PlanStorage( { children, className, siteId } ) {
...
 			mediaStorage.max_storage_bytes = 1024 * 1024 * 1024;
 		}
+
 		if ( sitePlanSlug === PLAN_WPCOM_PRO ) {
 			mediaStorage.max_storage_bytes = 50 * 1024 * 1024 * 1024;
 		}
+
+		if ( sitePlanSlug === PLAN_WPCOM_STARTER ) {
+			mediaStorage.max_storage_bytes = 6 * 1024 * 1024 * 1024;
+		}
```
Comment (obenland) [inline client/blocks/plan-storage/index.jsx:71]: I'm currently working on this file in #63394 as part of moving plan checks to feature checks (p4TIVU-a66-p2). 

Since `sitePlanSlug` is based on the current plan of the site and this is not about what features a plan has in general, do you think it would make sense to use a feature check here?

Or, if we don't want to show the highest available storage size for the site but specifically the storage site that comes with the current plan, could `planHasFeature` be used here to decouple the logic from checking specific plans?

File: packages/calypso-products/src/plans-list.tsx [hunk 15]
```diff
@@ -1574,6 +1576,25 @@ PLANS_LIST[ PLAN_P2_FREE ] = {
...
 // Brand new WPCOM plans
+PLANS_LIST[ PLAN_WPCOM_STARTER ] = {
+	...getDotcomPlanDetails(),
+	group: GROUP_WPCOM,
+	type: TYPE_STARTER,
+	term: TERM_ANNUALLY,
+	getTitle: () => i18n.translate( 'WordPress Starter' ),
+	getProductId: () => 1033,
+	getStoreSlug: () => PLAN_WPCOM_STARTER,
+	getPathSlug: () => 'starter-plan',
```
Comment (DavidRothstein) [inline packages/calypso-products/src/plans-list.tsx]: Drive-by comment -- this should just be `starter`, actually (that's how we want it to work elsewhere)?

Comment (chriskmnds) [reply packages/calypso-products/src/plans-list.tsx]: Thanks, will update. Assuming this is safe right? I'm not too familiar with most of these things. I think I was trying to connect the dots with your comment here 711-gh-Automattic/martech#issuecomment-1114502508

File: client/blocks/plan-storage/index.jsx (hunk 14 above)
Comment (obenland) [reply client/blocks/plan-storage/index.jsx:71]: @chriskmnds @DavidRothstein Is there any advice or thoughts you have for me on this?

Comment (chriskmnds) [reply client/blocks/plan-storage/index.jsx:71]: Apologies @obenland . I was planning to get back to you, but got caught up in a hectic release/revert scenario with this PR.

I agree if we can elevate this to feature checks would be cleaner. Also not a fun of hardcoding feature values that happens here and elsewhere (the legacy storage limit of 3GB is added in the logic inline here, also in the plan storage  feature for the grid).

> if we don't want to show the highest available storage size for the site but specifically the storage site that comes with the current plan

This is not too clear to me (whether this is what we want to do or not). The code renders the available space seen in the media library, so it might be conceptualy, but maybe not techically right now ❔  a site-specific characteristic

I will make a comment in the p2 post. I wanted to ask why Calypso has so much knowledge of features and feature details offhand, when all of that could (should?) be coming from the API - for all i know, i was able to render the complete Starter plan features list (albeit incomplete plan spec, with missing prices) without having a Starter plan returned from the REST endpoint. That seemed wrong to me, so anything in direction of consolidating looks like a win. (but of course, fear of sounding naive, cause a lot of missing pieces for me around these parts)

Comment (obenland) [reply client/blocks/plan-storage/index.jsx:71]: > I wanted to ask why Calypso has so much knowledge of features and feature details offhand, when all of that could (should?) be coming from the API - for all i know, i was able to render the complete Starter plan features list (albeit incomplete plan spec, with missing prices) without having a Starter plan returned from the REST endpoint. That seemed wrong to me, so anything in direction of consolidating looks like a win.

I don't _know_ but that doesn't keep me from assuming :) I could see it having started out as a way to power the plan comparison page with feature information, which then developed in the source of truth for plan <=> feature association for Calypso.

We'd _love_ to have that be powered by the features endpoint eventually, so everything that happens in `wpcom-features` automatically is available in Calypso as well.

Comment (chriskmnds) [reply client/blocks/plan-storage/index.jsx:71]: > We'd love to have that be powered by the features endpoint eventually, so everything that happens in `wpcom-features` automatically is available in Calypso as well.

Absolutely. Thanks for reply. Here's a bit more surface on the issue from a similar angle (where these feature descriptions/meta are used - and potentially in more places that I missed): https://github.com/Automattic/wp-calypso/issues/63764
